examples/machine_learning/src/lib.rs

```diff
@@ -138,63 +138,111 @@ fn evaluate_prediction(hits: &mut u32, animal: &Animal, prediction: &[f64]) -> (
     (actual_type, accurate)
 }
 
-struct Node;
+struct Node {
+    training_set_size: u32,
```

Comment:
May as well make these of type `usize` (since you use them to index into arrays later on, e.g. via https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.take ).

Reply:
I intended to, but was matching the existing code. :-)

Reply:
Done (however, causes a potentially bad cast later so dunno if this is actually better).